databricks_cli/secrets/cli.py

```diff
@@ -0,0 +1,312 @@
+# Databricks CLI
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+from tabulate import tabulate
+
+from databricks_cli.click_types import OutputClickType, SecretScopeClickType, SecretKeyClickType, \
+    SecretPrincipalClickType
+from databricks_cli.secrets.api import SecretApi
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS, pretty_format, truncate_string, \
+    error_and_quit, is_base64_str
+from databricks_cli.configure.config import provide_api_client, profile_option
+from databricks_cli.version import print_version_callback, version
+
+
+SCOPE_HEADER = ('Scope', 'Backend')
+SECRET_HEADER = ('Key name', 'Last updated')
+ACL_HEADER = ('Principal', 'Permission')
+VALUE_OPTIONS = ['string-value', 'bytes-value']
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help="Creates a secret scope.")
+@click.option('--scope', required=True, type=SecretScopeClickType(), help=SecretScopeClickType.help)
+@click.option('--initial-manage-principal',
+              help='The initial principal that can manage the created secret scope.')
+@profile_option
+@eat_exceptions
+@provide_api_client
+def create_scope(api_client, scope, initial_manage_principal):
+    """
+    Creates a new secret scope with given name.
+
+    If "initial_manage_principal" is specified, the initial ACL applied to the scope is
+    applied to the supplied principal (user or group) with MANAGE permissions.
+    The only supported principal for this option is the group "users", which
+    contains all users in the workspace. If "initial_manage_principal" is not specified,
+    the initial ACL with MANAGE permission applied to the scope is assigned to the
+    API request issuer's user identity.
+    """
+    if initial_manage_principal is not None and initial_manage_principal != "users":
```

Comment:
CLI shouldn't maintain this check. We should expect server return the right error message and we just forward it to users. This helps when we support other principals on the server side while users' CLI is not or cannot be updated.